### PR TITLE
website: Revise the "Module Sources" page

### PR DESCRIPTION
**Pull request by apparentlymart**
It's been a while since we made any significant updates to this page. The main theme of this revamp is to ensure that we highlight how to provide "ambient credentials" for each of the module sources using the standard mechanisms for each source type.

While there, I also attempted a general cleanup to highlight the main cases and make this less like a detailed description of all of `go-getter`'s esoteric features, and did some general copy-editing to write
it in our usual documentation "voice".

### CODE REVIEW

File: website/docs/modules/sources.html.markdown [hunk 1]
```diff
@@ -2,19 +2,23 @@
...
+for the desired child module.
 
-The `source` parameter tells Terraform where the module can be found.
-Terraform manages modules for you: it downloads them, organizes them on disk, checks for updates, etc. Terraform uses this `source` parameter to determine where it should retrieve and update modules from.
+Terraform uses this during the module installation step of `terraform init`
+to download the source code to a directory on local disk so that it can be
+used by other Terraform commands.
 
-Terraform supports the following sources:
+The module installed supports installation from a number of different source
```
Comment (mildwonkey) [inline website/docs/modules/sources.html.markdown]: do you mean module installer here?

File: website/docs/modules/sources.html.markdown [hunk 2]
```diff
@@ -60,243 +96,284 @@ module "consul" {
...
-**Note:** The double-slash, `//`, is important. It is what tells Terraform that this is the separator for a subdirectory, and not part of the repository itself.
+The value of the `ref` argument can be any reference that would be accepted
+by the `git checkout` command, including branch and tag names.
 
-Bitbucket URLs will require that Git or Mercurial is installed on your system, depending on the type of repository.
+## Generic Mercurial Repository
 
-## Private Bitbucket Repos
-Private bitbucket repositories must be specified similar to the [Generic Git Repository](#generic-git-repository) section below.
+You can use arbitrary Git repositories by prefixing the address with the
```
Comment (mildwonkey) [inline website/docs/modules/sources.html.markdown]: s/Git/Mercurial

**Comment by mildwonkey** [comment]
I had a couple of tiny questions, but this looks good.

My only other concern is with the location of the section on Modules in Package Sub-directories. I can only speak from personal experience but as a user, that was something I had to look up multiple times. It feels buried at the bottom, especially as it comes after what I (again, personally) think as sections that the average user is less likely to be interested in.

I'm not necessarily suggesting it be moved, but it could help if there's a logical way to include a link to this section from the top of the page.

**Comment by apparentlymart** [comment]
Thanks for the feedback, @mildwonkey!

I had actually intended to include a link to that sub-directories section but forgot to do so before I committed the result. I've now added a paragraph for it to the end of the text that precedes all of the source-specific instructions.

**Comment by mildwonkey** [comment]
Yes, I like the changes @apparentlymart!

**Pull request merged**